LayoutKit/Layouts/TextViewLayout.swift

```diff
@@ -0,0 +1,177 @@
+                            configure: ((TextView) -> Void)? = nil) {
+        self.init(text: .unattributed(text),
+                  font: font,
+                  lineFragmentPadding: lineFragmentPadding,
+                  contentInset: contentInset,
+                  layoutMargins: layoutMargins,
+                  textContainerInset: textContainerInset,
+                  layoutAlignment: layoutAlignment,
+                  flexibility: flexibility,
+                  viewReuseId: viewReuseId,
+                  configure: configure)
+    }
+
+    public convenience init(attributedText: NSAttributedString,
+                            font: UIFont = defaultFont,
+                            lineFragmentPadding: CGFloat = 0,
+                            contentInset: UIEdgeInsets = .zero,
+                            layoutMargins: UIEdgeInsets = .zero,
+                            textContainerInset: UIEdgeInsets = .zero,
+                            layoutAlignment: Alignment = defaultAlignment,
+                            flexibility: Flexibility = defaultFlexibility,
+                            viewReuseId: String? = nil,
+                            configure: ((TextView) -> Void)? = nil) {
+        self.init(text: .attributed(attributedText),
+                  font: font,
+                  lineFragmentPadding: lineFragmentPadding,
+                  contentInset: contentInset,
+                  layoutMargins: layoutMargins,
+                  textContainerInset: textContainerInset,
+                  layoutAlignment: layoutAlignment,
+                  flexibility: flexibility,
+                  viewReuseId: viewReuseId,
+                  configure: configure)
+    }
+
+    // MARK: - Layout protocol
+
+    open func measurement(within maxSize: CGSize) -> LayoutMeasurement {
+        let fittedSize = textSize(within: maxSize)
+        return LayoutMeasurement(layout: self, size: fittedSize.decreasedToSize(maxSize), maxSize: maxSize, sublayouts: [])
+    }
+
+    open func arrangement(within rect: CGRect, measurement: LayoutMeasurement) -> LayoutArrangement {
+        let frame = alignment.position(size: measurement.size, in: rect)
+        return LayoutArrangement(layout: self, frame: frame, sublayouts: [])
+    }
+
+    // MARK: - private helpers
+
+    private func textSize(within maxSize: CGSize) -> CGSize {
+        let options: NSStringDrawingOptions = [
+            .usesLineFragmentOrigin
+        ]
+
+        var size: CGSize
+        let heightInset = (textContainerInset.top + textContainerInset.bottom) + (layoutMargins.top + layoutMargins.bottom)
+        let widthInset = (textContainerInset.left + textContainerInset.right) + (layoutMargins.left + layoutMargins.right) + lineFragmentPadding
+            + (contentInset.left + contentInset.right)
+        let textContainerMaxSize = CGSize(width: maxSize.width + widthInset, height: maxSize.height + heightInset)
+        switch text {
```

Comment:
This code looks very similar to LabelLayout.textSize. Can there be some code sharing?